core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -145,6 +135,40 @@ internal class CacheFallbackOperations
             return users;
         }
 
+        public static Dictionary<String, AdalUserInfo> RemoveAdalUser(ILegacyCachePersistance legacyCachePersistance,
+            string displayableId, string environment, string identifier)
+        {
+            Dictionary<String, AdalUserInfo> users = new Dictionary<String, AdalUserInfo>();
+            try
+            {
+                IDictionary<AdalTokenCacheKey, AdalResultWrapper> dictionary =
+                    AdalCacheOperations.Deserialize(legacyCachePersistance.LoadCache());
+
+                List<AdalTokenCacheKey> keysToRemove = new List<AdalTokenCacheKey>();
+                foreach (KeyValuePair<AdalTokenCacheKey, AdalResultWrapper> pair in dictionary)
+                {
+                    if ( environment.Equals(new Uri(pair.Key.Authority).Host) &&
+                         displayableId.Equals(pair.Key.DisplayableId) &&
+                         identifier.Equals(ClientInfo.CreateFromJson(pair.Value.RawClientInfo).ToUserIdentifier()) )
+                    {
+                        keysToRemove.Add(pair.Key);
+                    }
+                }
+
+                foreach (AdalTokenCacheKey key in keysToRemove)
+                {
+                    dictionary.Remove(key);
+                }
+
+                legacyCachePersistance.WriteCache(AdalCacheOperations.Serialize(dictionary));
+            }
+            catch (Exception ex)
+            {
+                CoreLoggerBase.Default.Warning("RemoveAdalUser falid due to Exception - " + ex);
```

Comment:
>ex [](start = 92, length = 2)

perhaps it should be more specific? Do we need the stack trace or is the error message enough?
Aside: @jennyf19 one thing to keep in mind for the error/exception handling work

Reply:
logging should be reviewed completely in core